### PR TITLE
feat(TDI-46964): Request retry mechanism for Google Drive.

### PR DESCRIPTION
**Pull request by dmytrogrygorenko**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-46964

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/973/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/973/)

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveUtils.java [hunk 1]
```diff
@@ -221,7 +222,27 @@ public String getResourceId(String query, String resourceName, String type) thro
 
     public String findResourceByName(String resource, String type) throws IOException {
         if (resource.contains(PATH_SEPARATOR)) {
-            return findResourceByPath(resource, type);
+            long backoffValue = 0;
+            while (backoffValue <= 4) {
```
Comment (clesaec) [inline components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveUtils.java:226]: Hard to see here that we have 3 tries.
Should be more readable to have : 
int tries = 0;
while (tries < 3) { 
...
   long timeToSleep = 1 << tries;
...

**Review by clesaec — DISMISSED** [review]
Just one suggestion to simplify code.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/974/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/974/)

File: components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveUtils.java (hunk 1 above)
Comment (dmytrogrygorenko) [reply components/components-googledrive/components-googledrive-runtime/src/main/java/org/talend/components/google/drive/runtime/GoogleDriveUtils.java]: Fixed.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/975/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/975/)

**Review by clesaec — DISMISSED** [review]
:+1:

**Review by clesaec — APPROVED** [review]
:+1:

**Pull request merged**